aderyn_core/src/lib.rs

```diff
@@ -52,12 +52,21 @@ where
 }
 
 fn run_auditor_mode(context: &WorkspaceContext) -> Result<(), Box<dyn Error>> {
-    get_auditor_detectors().par_iter_mut().for_each(|detector| {
-        let found = detector.detect(context).unwrap();
-        if found {
-            detector.print(context);
-        }
-    });
+    let auditors_with_instances = get_auditor_detectors()
+        .par_iter_mut()
+        .flat_map(|detector| {
+            let found = detector.detect(context).unwrap();
+            if found {
+                return Some((detector.title(), detector.instances()));
+            }
+            None
+        })
+        .collect::<Vec<_>>();
+
+    for (detector_name, instances) in auditors_with_instances {
+        BasicAuditorPrinter::print(&instances, &detector_name);
+    }
+
```

Comment:
Before: No potential to handle the printing of instances on multiple Contexts

Now: It is possible to run this on multiple contexys

Reply:
(this will help me merge into ICF later if you approve this)